### PR TITLE
Restream convenience callbacks

### PR DESCRIPTION
**Pull request by danielballan**
Related to https://github.com/NSLS-II/ophyd/pull/307

### CODE REVIEW

File: bluesky/broker_callbacks.py [hunk 1]
```diff
@@ -81,20 +81,37 @@ def f(name, stop_doc):
...
-        # For convenience, I'll rely on the broker to get Events.
-        header = db[uid]
-        events = get_events(header)
-        callback('start', start)
-        for d in descriptors:
-            callback('descriptor', d)
-        for e in events:
-            callback('event', e)
-        callback('stop', stop_doc)
+        return restream(db[uid], callback)
```
Comment (tacaswell) [inline bluesky/broker_callbacks.py]: I think the test for this is not hooked up to insert into mds.

**Comment by danielballan** [comment]
Revisiting this now, the new function, `make_restreamer` is so dumb it probably shouldn't even exist. But the refactor of  `post_run` is still worth doing. Thoughts?

**Comment by tacaswell** [comment]
It might be a useful shim layer to keep around  Could also be used as a decorator.

**Comment by tacaswell** [comment]
any reason not to merge this?

**Comment by danielballan** [comment]
Worth merging, I guess. Might save someone ~10 minutes of thinking in the future.

**Pull request merged**